src/Common/tests/System/IO/FileCleanupTestBase.cs

```diff
@@ -78,6 +78,6 @@ protected virtual void Dispose(bool disposing)
         /// <param name="memberName">The member name of the function calling this method.</param>
         /// <param name="lineNumber">The line number of the function calling this method.</param>
         protected string GetTestFileName(int? index = null, [CallerMemberName] string memberName = null, [CallerLineNumber] int lineNumber = 0) =>
-            string.Format(index.HasValue ? "{0}_{1}_{2}" : "{0}_{1}", memberName ?? "TestBase", lineNumber, index.GetValueOrDefault());
+            string.Format(index.HasValue ? "{0}_{1}_{2}_{3}" : "{0}_{1}_{3}", memberName ?? "TestBase", lineNumber, index.GetValueOrDefault(), Guid.NewGuid().ToString("N"));
```

Comment:
A full guid might cause problems with path lengths in some of the tests, particularly on desktop runs. Perhaps Path.GetRandomFileName() would be good enough?